packages/cms-lib/lib/__tests__/config.js

```diff
@@ -73,4 +111,138 @@ describe('lib/config', () => {
       expect(fs.unlinkSync).toHaveBeenCalled();
     });
   });
+
+  describe('getAndLoadConfigIfNeeded)', () => {
+    beforeEach(() => {
+      setConfig(null);
+      process.env = {};
+    });
+
+    it('loads a config from file if no combination of environment variables is sufficient', () => {
+      const spy = jest.spyOn(configModule, 'getConfigPath');
+
+      getAndLoadConfigIfNeeded();
+      expect(configModule.getConfigPath).not.toHaveBeenCalled();
+      spy.mockRestore();
+    });
+
+    describe('oauth environment variable config', () => {
+      const {
+        portalId,
+        auth: {
+          clientId,
+          clientSecret,
+          tokenInfo: { refreshToken },
+        },
+      } = OAUTH2_CONFIG;
+      let portalConfig;
+
+      beforeEach(() => {
+        process.env = {
+          HUBSPOT_PORTAL_ID: portalId,
+          HUBSPOT_CLIENT_ID: clientId,
+          HUBSPOT_CLIENT_SECRET: clientSecret,
+          HUBSPOT_REFRESH_TOKEN: refreshToken,
+        };
+        getAndLoadConfigIfNeeded();
+        portalConfig = getConfig().portals[0];
```

Comment:
I think it is worthwhile to test the API that we use, so `getPortalConfig()` instead of reaching into `portals`.